tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1194 @@
+        self,
+    ):
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES,
+            None,
+            None,
+        )
+        with patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._network_interfaces_are_named_eth0_and_eth1",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._dns_configured",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._network_interfaces_config_files_exist",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._netplan_installed",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=True,
+        ), patch(
+            "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator._networking_service_enabled",  # noqa: E501
+            new_callable=PropertyMock,
+            return_value=False,
+        ), patch.object(
+            agw_network_configurator, "_update_interfaces_names_in_cloud_init"
+        ) as update_interfaces_names_in_cloud_init, patch.object(
+            agw_network_configurator, "_configure_grub"
+        ) as configure_grub, patch.object(
+            agw_network_configurator, "_update_grub_cfg"
+        ) as update_grub_cfg, patch.object(
+            agw_network_configurator, "_configure_dns"
+        ) as configure_dns, patch.object(
+            agw_network_configurator, "_create_interfaces_config_files"
+        ) as create_interfaces_config_files, patch.object(
+            agw_network_configurator, "_remove_netplan"
+        ) as remove_netplan, patch.object(
+            agw_network_configurator, "_enable_networking_service"
+        ) as enable_networking_service:
+            agw_network_configurator.configure_network_interfaces()
+        update_interfaces_names_in_cloud_init.assert_called_once()
+        configure_grub.assert_called_once()
+        update_grub_cfg.assert_called_once()
+        configure_dns.assert_called_once()
+        create_interfaces_config_files.assert_called_once()
+        remove_netplan.assert_called_once()
+        enable_networking_service.assert_called_once()
+
+    def test_given_agw_installer_network_configurator_when_configure_network_interfaces_is_called_and_interfaces_are_named_correctly_then_network_configuration_flow_is_executed_without_renaming_interfaces(  # noqa: E501
```

Comment:
Same comment, validating those flows is not really want we want to do. We don't want to validate that method x was called, we want to validate that DNS was configured.

Reply:
Same goes for most tests that validate flows